lib/handlers.py

```diff
@@ -278,18 +291,55 @@ def _run_auto_complete(view):
             'next_completion_if_showing': False,
         })
 
+    def _flatten_completions(self, completions, nesting=0):
+        if not completions:
+            return []
+
+        result = []
+        for c in completions:
+            result.append((self._brand_completion('  ' * nesting + c['display'], c['hint']),
+                           self._placeholder_text(c)))
+            if 'children' in c:
+                result += self._flatten_completions(c['children'], nesting + 1)
```

Comment:
I believe using `result.extend(self._flatten_completions(c['children'], nesting + 1))` should be more performant here.

Reply:
Good point, I'll change it